### PR TITLE
Add Flagd-ui to demo docs

### PR DESCRIPTION
**Pull request by maindotmarcell**
# Changes

Changes required for the documentation concerning the flagd-ui feature. 

As we have a [ready PR](https://github.com/open-telemetry/opentelemetry-demo/pull/1725) for the first version of the flagd-ui, we would like to start thinking about how this new feature can be included in the documentation. Below you can find what I've implemented in this PR as my suggestions:

- Screenshots -> Add screenshots of the UI
- Docker deployment -> Add :8080/feature route
- Kubernetes deployment -> Add :8080/feature route
- Feature flags page -> Mention the possibility to toggle the flags through the UI
- Requirements/architecture -> mention under feature flagging section

As it is not directly a microservice of the astronomy shop app, but it is an instrumented service nonetheless, there a few where it might make sense to include it (the flagd service itself is currently not included in these):
- Architecture charts
- List of Services

**Additionally it might make sense to create a separate docs page for this feature at some point.**

Thank you in advance for reviewing! 🚀 

Let me know if I missed anything :)

### CODE REVIEW

File: content/en/docs/demo/feature-flags.md
Comment (svrnm) [inline content/en/docs/demo/feature-flags.md:53]:
```suggestion
- **Basic View**: A user friendly view in which default variants (the same options that need to be changed when configuring through the
raw file) can be selected and saved for each feature flag. Currently, the basic view does not support fractional targeting.
- **Advanced View**: A view in which the raw configuration JSON file is loaded and can be edited within the browser. The view provides the
flexibility that comes with editing a raw JSON file, however it also provides schema checking to ensure that the
JSON is valid and that the provided configuration values are correct.
```

Comment (maindotmarcell) [reply content/en/docs/demo/feature-flags.md]: Done

File: content/en/docs/demo/requirements/architecture.md
Comment (maindotmarcell) [reply content/en/docs/demo/requirements/architecture.md]: Done

File: content/en/docs/demo/feature-flags.md
Comment (maindotmarcell) [reply content/en/docs/demo/feature-flags.md]: Done

File: content/en/docs/demo/feature-flags.md
Comment (maindotmarcell) [reply content/en/docs/demo/feature-flags.md]: Done

File: content/en/docs/demo/feature-flags.md
Comment (maindotmarcell) [reply content/en/docs/demo/feature-flags.md]: Done

**Comment by svrnm** [comment]
> > Thank you, I commented with a few change requests
> 
> @svrnm Thank you for the review, I've applied the suggested changes in the last commit

thanks!

@open-telemetry/demo-approvers please take a look!

File: content/en/docs/demo/feature-flags.md
Comment (julianocosta89) [inline content/en/docs/demo/feature-flags.md:40]: Can we move that up?

Ideally before the sentence:

```
These flags are managed by [`flagd`](https://flagd.dev), a simple
feature flag service that supports [OpenFeature](https://openfeature.dev). Flag
values are stored in the `src/flagd/demo.flagd.json` file. To enable a flag,
change the `defaultVariant` value in the config file for a given flag to "on".
```

Maybe even get rid of the whole setence 

```
Flag values are stored in the `src/flagd/demo.flagd.json` file.
```

**Review by julianocosta89 — COMMENTED** [review]
LGTM, we just need to get the [Helm chart PR](https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1367) merged before merging this one to avoid user's confusion.

Also, could you add the `flagd-ui` to the architecture page?
https://github.com/open-telemetry/opentelemetry.io/blob/main/content/en/docs/demo/architecture.md

I have it updated here if you would like: [mermaid.live](https://mermaid.live/edit#pako:eNqVlltvmzAUgP8KcqUpldIOQkiATZWqpN2qbVLW5mmlD45tCAuxmTFZWdv_PptbIIFMy0PgHH8-ts8NvwDEMAEuCDiM19py7tEkXRXCA-G7EBFtHkKp2HoUIsRSKkIaJMXQ4LrWVPS567qYCUqE5HHN4eb4T7iDHkUQrcnjYKYeH1f8_dW7s7HOCQ4T-WKcPymCi8rCTL53rSEnow1L91wpN9mARZAGkk05JxRlNVvKTRbFsUfJFoZRRd0ooYnwdJV51I9ggAe36v8iDZVeZDFJEA9juS2fwxRjIggSIaOVqVul1eaVuml0w0QUUjWRUUGotFy-HFsu9DFnz1lNaQslarUfb-iOZbkfqzOFWxgQOWkXYsIHd0pSk3JxP48GIX1uzYsYxAGhhEPB-ODxqxS1T5X8pLA4E2smtx7DbEtoHYlFIR4GvjqH3ApOkUBQwIjVCbUotNqsUHdF8VfKBKn470poUvFa7pkTxLZycQyb3r9vaVuTyhMk6zCOG-n9UMrt6CdCbYKkMnnzR-28L9DfwCp5PXonI8NlkmoXF1evn5fLxavWCt5BLDuwE4TKu4Pg_P86rZxQW87PkwPLmRw_KvgjojPPvUYe52hwv5i9ao1y7h5vl6eyclDd3caUVsu7yakJnfl2coXD7XShhR-b_eLkHloVcoo8yMRuNI9AHpAej9cNuHO0xyMNcu-L9uptqKY6y-7fCx7VXcOzzWJXZF4U0hsRTJI58bXiM6D5YRS5Z8bUnuj6ELGIcff3OhTkQwOV3azkfHO8muI-jnDVZ0p0ZZuOY_ehRUsqUV2HGFfoKoJo00RV66ts6tOR4fTZ3PfIarcG0S3YZ7f4bpSoNdFX0OizLFtjyY19CztWL5d3wxI1renouhdVn8ISnOqGZUz6waQ6DybQssd959l_6kqcOPbeboGDIdgSLmsOy4vLi0c1zQNiTbbEA658xcSHaSQ84NE3icJUsIeMIuAKnpIh4CwN1sD1oazYIUhjma2kvOJUiLyEyH76rbgZ5RekIYgh_cFYjUgRuC_gGbimdTlxnOlExtQ2TNMeTYYgA-7Isi8nI3MyHTm6_DnjtyH4kxvQL23HNA1dHxmmYzjT0fjtL7I5cro)

File: content/en/docs/demo/feature-flags.md
Comment (julianocosta89) [reply content/en/docs/demo/feature-flags.md:40]: I'm open to suggestions, but I'd say that it is more important to the user, how to access and change the feature flags than what file is being changed and what technology is being used.

I'd restructure to keep all the important information at the top of the page, and talk about the file and flagd at the bottom of it. As most users just want to know how to change feature flags.

But again, I'm opened to suggestions

**Review by julianocosta89 — COMMENTED** [review]
One minor suggestion, renaming the title and moving the initial sentence.

File: content/en/docs/demo/feature-flags.md
Comment (maindotmarcell) [reply content/en/docs/demo/feature-flags.md:15]: done

File: content/en/docs/demo/feature-flags.md
Comment (maindotmarcell) [reply content/en/docs/demo/feature-flags.md]: done

**Comment by julianocosta89** [comment]
LGTM, just waiting on https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1367

**Review by julianocosta89 — APPROVED** [review]

**Comment by julianocosta89** [comment]
@svrnm this one should be good to go.
I've fixed the failing checks and that added some extra files to the PR (unrelated to the PR itself).
Happy to revert if that is not the way to do it.

**Review by svrnm — APPROVED** [review]

**Pull request merged**
